src/Gaufrette/Adapter/AwsS3.php

```diff
@@ -111,10 +111,12 @@ public function rename($sourceKey, $targetKey)
 
         try {
             $this->service->copyObject($options);
-            return true;
         } catch (\Exception $e) {
             return false;
         }
+        $this->delete($sourceKey);
```

Comment:
I have no idea what the author of the lib thinks, so lets hear for their opinion.

the initial version and also the new changed form both make sense for me.